packages/monaco-editor-react/README.md

```diff
@@ -0,0 +1,135 @@
+
+The `monaco-languageclient` can be activated to connect to a language server either via jsonrpc over a websocket to an exernal server process or via language server protocol for browser where the language server runs in a web worker.
+
+## Getting Started
+
+If you have node.js LTS available, then from the root of the project run:
+
+```bash
+npm i
+npm run build
+```
+
+Afterwards launch the Vite.js development mode:
+
+```bash
+npm run dev
+```
+
+You find examples (manual human testing) in the root of the repository [index.html](../../index.html). They can be used once Vite is running.
+
+## Usage
+
+You can import the monaco react component for easy use in an existing React project. Below you can see a quick example of a fully functional implementation for some TypeScript.
+
+```ts
+import { MonacoEditorReactComp, addMonacoStyles } from 'monaco-editor-react/allLanguages';
+
+addMonacoStyles('monaco-editor-styles');
+
+const languageId = 'typescript';
+
+const codeMain = `function sayHello(): string {
+    return "Hello";
+};`;
+
+const comp = <MonacoEditorReactComp
+    languageId={languageId}
+    text={codeMain}
+    style={{
+        'paddingTop': '5px',
+        'height': '100%',
+        'width': '100%'
+    }}
+/>;
+```
+
+You can also pass in custom syntax highlighting using the Monarch language. Here's an example for a simple statemachine language.
+
+```ts
+// helps to bring in Monaco from the wrapper, but not required
+import { monaco } from 'monaco-editor-wrapper';
+
+const syntaxHighlighting = {
+  keywords: [
+      'actions', 'commands', 'end', 'events', 'initialState', 'state', 'statemachine'
+  ],
+  tokenizer: {
+      root: [
+          [/[a-z_$][\w$]*/, {
+              cases: {
+                  '@keywords': 'keyword',
+                  '@default': 'identifier'
+              }
+          }],
+          { include: '@whitespace' }
+      ],
+      comment: [
+          [/[^\/*]+/, 'comment'],
+          [/\/\*/, 'comment', '@push'],
+          ["\\*/", 'comment', '@pop'],
+          [/[\/*]/, 'comment']
+      ],
+      whitespace: [
+          [/[ \t\r\n]+/, 'white'],
+          [/\/\*/, 'comment', '@comment'],
+          [/\/\/.*$/, 'comment'],
+      ]
+  }
+} as monaco.languages.IMonarchLanguage;
+
+...
+
+<MonacoEditorReactComp
+    languaegId="statemachine"
+    text={codeMain}
+    syntax={syntaxHighlighting}>
+```
+
+## Bundled Usage
+
+For special cases where you might want the component to be processed in advance, so we also provide a pre-bundled version that you can reference instead. This can be helpful if you're working within some other framework besides React (Hugo for example).
+
+```ts
+import { MonacoEditorReactComp } from '@typefox/monaco-editor-react/bundle';
+
+...
+
+const comp = <MonacoEditorReactComp languageId="statemachine" text={codeMain}/>
+```
+
+## Invoking Custom Commands
```

Comment:
We should describe this for the actual wrapper, too. But not in the context of this PR. Let's merge it first.